### PR TITLE
Move Smallrye JWT Build docs to a separate guide

### LINKED ISSUE #20736: Split the JWT guide into JWT and JWT Build/generate guides
### Description

The JWT guide,  https://quarkus.io/guides/security-jwt, like its OIDC friends, has become badly structured (Quickstart and the ref info is intermixed) but it is worse than that with the JWT guide as the section on generating JWT tokens is totally unrelated - it can be used alongside MP JWT servers but it is meant not to be tied in to MP JWT, so the section about generating JWT tokens has to be moved into its own guide to make it more visible.
I'd still like to avoid splitting the actual quickstart and ref parts along the way - in this case alone it would produce 4 docs instead of 1 (both JWT server and JWT build may have the quickstarts and the ref parts, so alongside with OIDC we'd have 4 here + 2 per OIDC service and 2 per OIDC web-app - 8 in total instead of 3 and the sec guides will start exploding). This is certainly something that needs to be seriously looked at, but as I suggested in the OIDC doc issue, I'd like to try to do it stages, first improve the structure and then move on to a split - I think introducing a hierarchical view may simplify it. 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #20736

This PR moves `smallrye-jwt-build` related docs into a dedicated guide and expands it further with more details - it should be done to decouple it from the `smallrye-jwt` guide to make it more obvious the provided API can be used independently of MP JWT and also clean up the `smallrye-jwt` guide.

`smallrye-jwt` docs's structure has also been aligned to the the OIDC doc structure.

I'll follow up with the Quarkus guides update on the web site once this one is merged.

CC @gastaldi @stuartwdouglas

### CODE REVIEW

File: docs/src/main/asciidoc/security-jwt.adoc [hunk 1]
```diff
@@ -631,27 +633,7 @@ worked through in this quickstart guide as well as some additional endpoints tha
...
-|mp.jwt.verify.publickey.location|none|Config property allows for an external or internal location of Public Key to be specified.  The value may be a relative path or a URL. If the value points to an HTTPS based JWK set then, for it to work in native mode, the `quarkus.ssl.native` property must also be set to `true`, see link:native-and-ssl[Using SSL With Native Executables] for more details.
-|mp.jwt.verify.publickey.algorithm|`RS256`|Signature algorithm. Set it to `ES256` to support the Elliptic Curve signature algorithm.
-|mp.jwt.decrypt.key.location|none|Config property allows for an external or internal location of Private Decryption Key to be specified.
-|mp.jwt.verify.issuer|none|Config property specifies the value of the `iss` (issuer) claim of the JWT that the server will accept as valid.
-|mp.jwt.verify.audiences|none|Comma separated list of the audiences that a token `aud` claim may contain.
-|mp.jwt.token.header|`Authorization`|Set this property if another header such as `Cookie` is used to pass the token.
-|mp.jwt.token.cookie|none|Name of the cookie containing a token. This property will be effective only if `mp.jwt.token.header` is set to `Cookie`.
-|===
```
Comment (gastaldi) [inline docs/src/main/asciidoc/security-jwt.adoc:654]: Isn't this generated in a config file already?

File: docs/src/main/asciidoc/security-jwt.adoc [hunk 2]
```diff
@@ -947,154 +902,76 @@ public class ProtectedResource {
...
-|smallrye.jwt.encrypt.key.id|`none`|Encryption key identifier which is checked only when JWK keys are used.
-|smallrye.jwt.new-token.lifespan|300|Token lifespan in seconds which will be used to calculate an `exp` (expiry) claim value if this claim has not already been set.
-|smallrye.jwt.new-token.issuer|none|Token issuer which can be used to set an `iss` (issuer) claim value if this claim has not already been set.
-|smallrye.jwt.new-token.audience|none|Token audience which can be used to set an `aud` (audience) claim value if this claim has not already been set.
-|smallrye.jwt.new-token.override-matching-claims|false| Set this property to `true` for `smallrye.jwt.new-token.issuer` and `smallrye.jwt.new-token.audience` values to override the already initialized `iss` (issuer) and `aud` (audience) claims.
+|smallrye.jwt.verify.key.location|NONE|Location of the verification key which can point to both public and secret keys. Secret keys can only be in the JWK format. Note that 'mp.jwt.verify.publickey.location' will be ignored if this property is set.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/security-jwt.adoc:950]: Isn't this content generated in a separate config ADOC file already?

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-jwt.adoc:950]: Hi @gastaldi it should be correct, `smallrye.jwt.verify.key.location`/etc are specific to `smallrye-jwt` only, `smallrye.jwt.new-token*` are used by `smallrye-jwt-build` so they got moved

Comment (sberyozkin) [reply docs/src/main/asciidoc/security-jwt.adoc:950]: @gastaldi I've double checked `smallrye-jwt-build` and `smallrye-jwt` docs reference only their specific properties as far as I can see

**Review by gastaldi — APPROVED** [review]

**Comment by sberyozkin** [comment]
Thanks @gastaldi will follow with a website PR now

**Comment by sberyozkin** [comment]
Thanks @geoand as well, haven't noticed it was you who merged it :-)

**Comment by geoand** [comment]
YW :)

**Comment by sberyozkin** [comment]
Proposing a backport since it is really  a cleanup as opposed to a new doc guide/feature, but would not be a problem if it s not backported

**Pull request merged**